pkg/registry/core/pod/strategy.go

```diff
@@ -91,7 +92,12 @@ func (podStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 		QOSClass: qos.GetPodQOS(pod),
 	}
 
-	podutil.DropDisabledPodFields(pod, nil)
+	// Mirror pods should not have Generation set.
+	if metav1.HasAnnotation(pod.ObjectMeta, v1.MirrorPodAnnotationKey) {
```

Comment:
I thought we weren't going to do anything special in the apiserver for mirror pods - https://github.com/kubernetes/enhancements/pull/5068#discussion_r1945426421

Special-casing them here doesn't seem that useful

Reply:
ack, taken out